contrib/pkg/awsprivatelink/enable.go

```diff
@@ -0,0 +1,224 @@
+
+	openshiftcorev1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	"github.com/openshift/hive/pkg/constants"
+
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type enableOptions struct {
+	homeDir string
+	// AWS region of the Hive cluster
+	region        string
+	infraId       string
+	dnsRecordType string
+
+	kubernetesClient      *kubernetes.Clientset
+	hiveClient            *hiveclient.Clientset
+	openshiftConfigClient *openshiftcorev1.ConfigV1Client
+
+	// AWS EC2 client in Hive cluster's region
+	EC2Client *ec2.EC2
+}
+
+func NewEnableAWSPrivateLinkCommand() *cobra.Command {
+	opt := &enableOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "enable",
+		Short: "Enable AWS PrivateLink",
+		Long: `Enable AWS PrivateLink:
+1) Extract AWS Hub account credentials from the environment where this command is called
+2) Create a Secret with the above credential. The Secret is referred to by HiveConfig.spec.awsPrivateLink.credentialsSecretRef
+3) Add the active cluster's VPC to the list of HiveConfig.spec.awsPrivateLink.associatedVPCs`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := opt.Complete(cmd, args); err != nil {
+				return
+			}
+			if err := opt.Validate(cmd, args); err != nil {
+				return
+			}
+			if err := opt.Run(cmd, args); err != nil {
+				return
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&opt.dnsRecordType, "dns-record-type", "Alias", "HiveConfig.spec.awsPrivateLink.dnsRecordType")
+	return cmd
+}
+
+func (o *enableOptions) Complete(cmd *cobra.Command, args []string) error {
+	// Get client config
+	cfg, err := hiveutils.GetClientConfig()
+	if err != nil {
+		log.WithError(err).Fatal("Failed to generate client config")
+	}
+
+	// Get Kubernetes client
+	kubernetesClient, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		log.WithError(err).Fatal("Failed to create Kubernetes clientset")
+	}
+	o.kubernetesClient = kubernetesClient
+
+	// Get Hive client
+	hiveClient, err := hiveclient.NewForConfig(cfg)
+	if err != nil {
+		log.WithError(err).Fatal("Failed to create Hive clientset")
+	}
+	o.hiveClient = hiveClient
+
+	// Get Openshift CoreV1 client
+	openshiftConfigClient, err := openshiftcorev1.NewForConfig(cfg)
+	if err != nil {
+		log.WithError(err).Fatal("Failed to create Openshift Config/v1 client")
+	}
+	o.openshiftConfigClient = openshiftConfigClient
+
+	// Get current user's home directory
+	o.homeDir = "."
+	u, err := user.Current()
+	if err != nil {
+		log.WithError(err).Fatal("Failed to get the current user's home directory")
+	}
+	o.homeDir = u.HomeDir
+
+	// Get EC2 client
+	o.region, o.infraId = o.getRegionAndInfraId()
+	log.Debugf("Found region = %v, infraId = %v", o.region, o.infraId)
+	o.EC2Client = awsutils.GetAWSEC2Client(o.region)
```

Comment:
As noted below, please use the existing AWS client wrapper. You should be able to init the client via [client.New()](https://github.com/openshift/hive/blob/4dfe9ecb15e1db492177e12ab1d5385b98092850/pkg/awsclient/client.go#L397) or one of its brethren in that file. In your case possibly [NewClientFromSecret()](https://github.com/openshift/hive/blob/4dfe9ecb15e1db492177e12ab1d5385b98092850/pkg/awsclient/client.go#L482) with (confusingly) a `nil` Secret.